db/dcrsqlite/apisource.go

```diff
@@ -1312,23 +1312,46 @@ func (db *wiredDB) GetExplorerAddress(address string, count, offset int64) *expl
 		TotalUnspent: int64(totalreceived - totalsent),
 	}
 	return &explorer.AddressInfo{
-		Address:           address,
-		MaxTxLimit:        maxcount,
-		Limit:             count,
-		Offset:            offset,
-		NumUnconfirmed:    numUnconfirmed,
-		Transactions:      addressTxs,
-		NumTransactions:   numTxns,
-		NumFundingTxns:    numReceiving,
-		NumSpendingTxns:   numSpending,
-		AmountReceived:    totalreceived,
-		AmountSent:        totalsent,
-		AmountUnspent:     totalreceived - totalsent,
-		Balance:           balance,
-		KnownTransactions: numberMaxOfTx,
-		KnownFundingTxns:  numReceiving,
-		KnownSpendingTxns: numSpending,
+		Address:            address,
+		MaxTxLimit:         maxcount,
+		Limit:              count,
+		Offset:             offset,
+		NumUnconfirmed:     numUnconfirmed,
+		Transactions:       addressTxs,
+		NumTransactions:    numTxns,
+		NumFundingTxns:     numReceiving,
+		NumSpendingTxns:    numSpending,
+		AmountReceived:     totalreceived,
+		AmountSent:         totalsent,
+		AmountUnspent:      totalreceived - totalsent,
+		Balance:            balance,
+		KnownTransactions:  numberMaxOfTx,
+		KnownFundingTxns:   numReceiving,
+		KnownSpendingTxns:  numSpending,
+		IsTechnicalAddress: isTechnicalAddress,
+	}
+}
+
+// IsZeroHashP2PHKAddress checks if given address is special
+// see https://github.com/decred/dcrdata/issues/358 for details
+func IsZeroHashP2PHKAddress(checkAddressString string, params *chaincfg.Params) bool {
+
+	//sanity check
+	checkAddress, err := dcrutil.DecodeAddress(checkAddressString)
```

Comment:
why sanity check at all?  You just need to see if the input string is equal to the zero-hash p2pkh.